keps/sig-cli/2227-kubectl-default-container/README.md

```diff
@@ -0,0 +1,279 @@
+- Add a warning deprecation message when using the annotation `kubectl.kubernetes.io/default-logs-container`
+
+#### Beta -> GA Graduation
+
+- Gather feedback from developers and surveys
+- At least 2 release cycles pass to gather feedback and bug reports during
+- The deprecation message of the annotation `kubectl.kubernetes.io/default-logs-container` will be removed and this annotation will stop working.
+
+### Upgrade / Downgrade Strategy
+
+If kubectl is upgraded and no annotation is found, nothing happens.
+If there's an annotation and kubectl is downgraded, the old behavior of using the first container will come back to the users.
+
+### Version Skew Strategy
+None
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism:
+      - This feature is explicitly opt-in since it need user to add specified
+        annotation in pod.
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+        - No. Disabling the feature would be a client behaviour.
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? 
+        - No. It is a client behaviour only.
+
+* **Does enabling the feature change any default behavior?**
+  - No. The old behavior is notification when no default container is specified.
+    Current behavior is cover that once specified annotation is in the pod spec.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  - Yes. When `--container` flag is specified, we will ignore
+    the specified annotation.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  - Nothing. It uses the first container as default in old behivior. It uses the 
+    annotation with this feature implemented, the first container will be used
+    if the annotation is not defined.
+
+* **Are there any tests for feature enablement/disablement?**
+  - There are unit tests in `staging/src/k8s.io/kubectl/pkg/cmd/exec/` and 
+    `staging/src/k8s.io/kubectl/pkg/polymorphichelpers/` that
+    verify the behaviour.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?  
+  - None
+* **What specific metrics should inform a rollback?
+  - None
+* **Were upgrade and rollback tested?
+  - None, or with adding and deleting the annotation from a pod to test it.
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.? Even if applying deprecation policies, they may still surprise some users. No.
+  - None
+
+### Monitoring Requirements
+None
+
+### Dependencies
```

Comment:
It's weird to see these empty. I thought about it a while, but since it's just an annotation used for client-side hint, I couldn't really think of any scalability or dependency problems.  Can you add the questions and indicate that so we know they were considered.